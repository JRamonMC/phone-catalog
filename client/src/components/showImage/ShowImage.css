.img-cont {
  width: 200px;
  height: 250px;
}