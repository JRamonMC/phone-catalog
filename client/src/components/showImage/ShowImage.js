import React from 'react';
import {API} from '../../config';
import './ShowImage.css';
import Loader from "react-loader-spinner";




const ShowImage = ({item, url}) => (
  
  <div className="product-img">
    {item._id ? <img
      src={`${API}/${url}phoneimage/${item._id}`}
      alt={item.name}
      className="mb-3 img-cont"
      style={{maxHeight: "600px", maxWidth:"300px"}}/> : 
      <Loader
        type="Puff"
        color="#00BFFF"
        height={100}
        width={100}
      />}
  </div>
)

export default ShowImage;