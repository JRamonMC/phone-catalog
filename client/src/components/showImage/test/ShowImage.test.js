import React from "react";
import ShowImage from "./../ShowImage";
import { shallow } from "enzyme";

describe("ShowImage test", () => {
  const fakeDataPhone = {
    _id: "id",
    name: "test",
    manufacturer: "test-m",
    description: "test-d",
    color: "test-c",
    price: "1",
    screen: "test-s",
    processor: "2",
    ram: "3",
    imageRaw: "test-i",
  };

  const fakeUrl = "fakeUrl";

  it("Show image render", () => {
    const wrapper = shallow(<ShowImage item={fakeDataPhone} url={fakeUrl} />);
    expect(wrapper).toMatchSnapshot();
  });
});
