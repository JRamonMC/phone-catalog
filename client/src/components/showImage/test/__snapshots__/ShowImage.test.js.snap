// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowImage test Show image render 1`] = `
<div
  className="product-img"
>
  <img
    alt="test"
    className="mb-3 img-cont"
    src="http://localhost:5000/api/phone/fakeUrlphoneimage/id"
    style={
      Object {
        "maxHeight": "600px",
        "maxWidth": "300px",
      }
    }
  />
</div>
`;
